src/neo/Consensus/ConsensusService.cs

```diff
@@ -174,11 +177,20 @@ private void InitializeConsensus(byte viewNumber)
                 }
                 else
                 {
+                    // Too long (if changeview happens during consensus) or too short (which will occur during synchronization) 
+                    // consensus period will be filtered. Only normal consensus should be taken into account.
+                    if (block_received_gap > Blockchain.TimePerBlock / 3 && block_received_gap < Blockchain.TimePerBlock * 2)
```

Comment:
I think that we should take the average of the last X blocks, because otherwise, we only fix big delays

Reply:
Maybe we can have a try by using the last X blocks, currently we are using [EMA(exponential moving average)](https://en.wikipedia.org/wiki/Moving_average#Exponential_moving_average).